src/NuGet.Services.Metadata.Catalog.Monitoring/Status/PackageMonitoringStatusService.cs

```diff
@@ -66,20 +67,34 @@ public Task<PackageMonitoringStatus> GetAsync(FeedPackageIdentity package, Cance
             var packageStatuses = new List<PackageMonitoringStatus>();
 
             var storage = GetStorage(state);
-            foreach (var listItem in await storage.List(token))
-            {
-                var packageStatus = await GetPackageAsync(storage, listItem.Uri, token);
-
-                if (packageStatus == null)
+            return (await Task.WhenAll((await storage.List(token)).Select(listItem =>
+                Task.Run(async () =>
                 {
-                    _logger.LogWarning("Unable to get package status from {PackageUri}!", listItem.Uri);
-                    continue;
-                }
-
-                packageStatuses.Add(packageStatus);
-            }
-
-            return packageStatuses;
+                    PackageMonitoringStatus status = null;
+                    try
+                    {
+                        status = await GetPackageAsync(storage, listItem.Uri, token);
+                    }
+                    catch (Exception e)
+                    {
+                        _logger.LogWarning("Unable to get package status from {PackageUri}!", listItem.Uri);
+
+                        /// Construct a <see cref="PackageMonitoringStatus"/> from the <see cref="Uri"/> with this as the exception.
+                        var uriSegments = listItem.Uri.Segments;
+                        // The second to last segment is the id.
+                        var id = uriSegments[uriSegments.Length - 2].Trim('/');
+
+                        // The last segment is {id}.{version}.json.
+                        // Remove the id and the "." from the beginning.
+                        var version = uriSegments[uriSegments.Length - 1].Substring(id.Length + ".".Length);
+                        // Remove the ".json" from the end.
+                        version = version.Substring(0, version.Length - ".json".Length);
+
+                        status = new PackageMonitoringStatus(new FeedPackageIdentity(id, version), new StatusDeserializationException(e));
+                    }
+                    return status;
+                })
+            ))).Where(p => p != null);
```

Comment:
Lost my place... is 'p' the Task<TResult> status result from Task.Run? Maybe give 'p' a better name?